# Copyright (c) Meta Platforms, Inc. and affiliates.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,10 +62,8 @@
   target_compile_features(${target} ${visibility} cxx_std_20)
   # GCC 10 needs this even under -std=c++20, and folly keys FOLLY_HAS_COROUTINES
   # off it. Inert from GCC 11 on.
-  if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    target_compile_options(${target} ${visibility}
-                           $<$<COMPILE_LANGUAGE:CXX>:-fcoroutines>)
-  endif ()
+  target_compile_options(${target} ${visibility}
+                         $<$<COMPILE_LANG_AND_ID:CXX,GNU>:-fcoroutines>)
   target_include_directories(
     ${target} ${visibility} $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
     $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>)
